### DISCUSSION TITLE
Is it possible to speed up .stream()?

### DISCUSSION THREAD

**Discussion by r0hin** [sindresorhus/got#1837] [Q&A] [answered]
Hi. I'm using got on an express server where I want to serve a file from another URL. What I have is:
``` 
got.stream(url, {}).pipe(res).on('finish', () => {
    res.end();
})
```
However, it creates something like [this](https://parallelcloud.herokuapp.com/stream?url=0x68747470733a2f2f72322d2d2d736e2d7035716c736e646b2e676f6f676c65766964656f2e636f6d2f766964656f706c61796261636b3f6578706972653d313632383831323631342665693d356d41565961793445625751687762547161367741772669703d332e39302e3130352e38342669643d6f2d41463048706d687a637230466c7975375f716a43715037634735786d34597a34687a53556e4633756151714c26697461673d32353126736f757263653d796f7574756265267265717569726573736c3d796573266d683d7a58266d6d3d33312532433236266d6e3d736e2d7035716c736e646b253243736e2d76677173726e6c6b266d733d61752532436f6e72266d763d6d266d76693d3226706c3d313226696e697463776e646270733d39303632353026767072763d31266d696d653d617564696f2532467765626d266e733d46337267735045433337414c664353675546735478764147266769723d79657326636c656e3d33303039393534266475723d3139302e303031266c6d743d31363139343638313732363837333437266d743d3136323837383938303826667669703d32266b656570616c6976653d79657326666578703d3234303031333733253243323430303732343626633d574542267478703d35353332343334266e3d6d70587976366957324562685069353256782d302673706172616d733d65787069726525324365692532436970253243696425324369746167253243736f757263652532437265717569726573736c253243767072762532436d696d652532436e73253243676972253243636c656e2532436475722532436c6d74266c73706172616d733d6d682532436d6d2532436d6e2532436d732532436d762532436d7669253243706c253243696e697463776e64627073266c7369673d414733435f78417752674968414e6f394b32454d4334536b76744b6e775079617a444550666f525a567956746c42437036546b6a6b63577a4169454139354d6a50775a72514371355f38746e7166384f3962547765587a586d786f346f397955446a2d5a44373025334426726174656279706173733d796573267369673d414f7130514a3877524149674e5a484278574c6f78366a743468452d704d326561512d334873695655515348586f4551785a726962506b4349414d6145324a335775396d72424755626a63446e41386e6e556e727779474543376f3465426b2d376e3945). Basically, the loaded amount goes up, but it does so in chunks and theres large gaps in between the download.

Is it possible to have the stream forwarded to the result as fast as possible? The file should only be a few megabytes. Also, when you first make a request, it seems like you're initially presented with the first x seconds of the audio file at a very fast speed but the remainder seems to be slow.

Alternatively, maybe 'seeking' support would be useful here?

Anyhow, thank you for your help and thanks for making this library! :)

**Comment by szmarczak** [comment]
> Is it possible to have the stream forwarded to the result as fast as possible?

It already does. You're limited by the environment only.

> theres large gaps in between the download.

Most probably caused by upstream or heroku. Have you tried running it locally?

**Reply by r0hin** [reply]
Yeah, I've tried it locally. It's almost the same thing exactly. For reference, going to the source directly and streaming/downloading it takes about 1 second while doing it through the stream, it takes about 20 seconds.

**Reply by r0hin** [reply]
> > Is it possible to have the stream forwarded to the result as fast as possible?
> 
> It already does. You're limited by the environment only.
> 
> > theres large gaps in between the download.
> 
> Most probably caused by upstream or heroku. Have you tried running it locally?

I did some more digging and found that it only happens on specific URLs. Here's an example URL: [link](https://r7---sn-cxaaj5o5q5-tt1e6.googlevideo.com/videoplayback?expire=1628835915&ei=67sVYeVMiIH-1w-MkJyAAg&ip=74.14.164.182&id=o-ABTB6CdkJnY0JnmvZ1p957saUas-cimK5MVQq4rQEmv2&itag=251&source=youtube&requiressl=yes&mh=zX&mm=31%2C26&mn=sn-cxaaj5o5q5-tt1e6%2Csn-vgqsrnlk&ms=au%2Conr&mv=m&mvi=7&pl=24&initcwndbps=1533750&vprv=1&mime=audio%2Fwebm&ns=oaEayCJMM9ca2WqDjLyf-gUG&gir=yes&clen=3009954&dur=190.001&lmt=1619468172687347&mt=1628814040&fvip=2&keepalive=yes&fexp=24001373%2C24007246&c=WEB&txp=5532434&n=UsFoyxz833uj2jUtagG&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgQqQlhPuO7Frni0N-AYxxRfw-N2GVM2ELaxGky9GG2IICIB2Y_kJEYHpmL77-S6_-tfGkgwTGOv-wneFqDHZrtUGj&ratebypass=yes&sig=AOq0QJ8wRgIhAJlRb2Ic3bA8ds1_bP8y3PihU6nPuLcLUoYc1NhSDRh6AiEAxQkDFaBs6Ho8lDIFWNXEd5r6fycPop9imfvJz500SOk%3D). Since the download is extremely fast on a regular browser, is it possible to simulate a browser with the library through NodeJS?

**Comment by r0hin** [comment, marked as answer]
Able to fix it with:
```
    headers: {
      "range": "bytes=0-",
    }
```
